### PR TITLE
async-iteration: add tests for AsyncGenerationFunction and AsyncIteratorPrototype built-in

### PR DESCRIPTION
**Pull request by spectranaut**

### CODE REVIEW

File: test/built-ins/AsyncGeneratorFunction/has-instance.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-properties-of-asyncgeneratorfunction
+description: >
+    AsyncGenerator function instances are correctly reported as instances of the
+    AsyncGeneratorFunction intrinsic.
+features: [async-iteration]
+---*/
+
+var AsyncGeneratorFunction = Object.getPrototypeOf(function* () {}).constructor;
```
Comment (leobalter) [inline test/built-ins/AsyncGeneratorFunction/has-instance.js]: missing `async`

File: test/built-ins/AsyncGeneratorFunction/has-instance.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+function* agDecl() {}
+var agExpr = function* () {};
```
Comment (leobalter) [inline test/built-ins/AsyncGeneratorFunction/has-instance.js]: missing async in the two lines above as well

File: test/built-ins/AsyncGeneratorFunction/instance-await-expr-in-param.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// YieldExpression is permitted in function body.
+AsyncGeneratorFunction('x = await 42');
+
+assert.throws(SyntaxError, function() {
+  AsyncGeneratorFunction('x = await 42', '');
+}, 'AwaitExpression not permitted in parameters');
+
+var withinAsyncGenerator = async function*() {
+  AsyncGeneratorFunction('x = await 42', '');
+};
```
Comment (leobalter) [inline test/built-ins/AsyncGeneratorFunction/instance-await-expr-in-param.js]: this is tricky, but nice to have tested. +1

Comment (spectranaut) [reply test/built-ins/AsyncGeneratorFunction/instance-await-expr-in-param.js]: I think you can credit to @jugglinmike for this tricky logic! He wrote the analogous test in GeneratorFunction directory (for yield) :)

**Comment by spectranaut** [comment]
Ok @leobalter I addressed your review in the last commit!

File: test/built-ins/AsyncGeneratorFunction/instance-await-expr-in-param.js (hunk 3 above)
Comment (leobalter) [reply test/built-ins/AsyncGeneratorFunction/instance-await-expr-in-param.js]: you could also add tests for the yield expressions here, maybe in a follow up PR?

File: test/built-ins/AsyncIteratorPrototype/Symbol.asyncIterator/prop-desc.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: sec-asynciteratorprototype
```
Comment (leobalter) [inline test/built-ins/AsyncIteratorPrototype/Symbol.asyncIterator/prop-desc.js]: use esid

File: test/built-ins/AsyncIteratorPrototype/Symbol.asyncIterator/return-val.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: sec-asynciteratorprototype-asynciterator
```
Comment (leobalter) [inline test/built-ins/AsyncIteratorPrototype/Symbol.asyncIterator/return-val.js]: use esid

**Comment by leobalter** [comment]
I rebased this branch with master and applied a fix for the typos. Should be good to land after the CI.

**Pull request merged**